File: cpp/1656.cpp
#include <string>
#include <unordered_map>

class OrderedStream {
private:
  int N;
  int pos;
  std::unordered_map<int, std::string> nodes;

  std::vector<std::string> flush() {
    std::vector<std::string> acc;

    while (nodes.contains(pos) && (pos <= N)) {
      acc.push_back(nodes.find(pos)->second);
      nodes.erase(pos);
      pos += 1;
    }

    return acc;
  }

public:
  OrderedStream(int n) : N(n), pos(1) {}

  std::vector<std::string> insert(int idKey, std::string value) {
    nodes.insert(std::make_pair(idKey, value));
    return flush();
  }
};

File: src/leetcode/1500-1749/1656.py
from typing import List

class OrderedStream:
    nodes: dict[int, str]
    pos: int
    n: int

    def __init__(self, n: int) -> None:
        self.nodes = {}
        self.pos = 1
        self.n = n

    def _flush_mut(self) -> list[str]:
        acc : list[str] = []

        while self.pos in self.nodes and self.pos <= self.n:
            acc.append(self.nodes.pop(self.pos))
            self.pos += 1

        return acc

    def insert(self, idKey: int, value: str) -> List[str]:
        self.nodes[idKey] = value
        return self._flush_mut()
